cartography/intel/aws/ec2/key_pairs.py

```diff
@@ -16,42 +17,44 @@
 
 @timeit
 @aws_handle_regions
-def get_ec2_key_pairs(boto3_session: boto3.session.Session, region: str) -> List[Dict]:
+def get_ec2_key_pairs(boto3_session: boto3.session.Session, region: str) -> list[dict[str, Any]]:
     client = boto3_session.client('ec2', region_name=region, config=get_botocore_config())
     return client.describe_key_pairs()['KeyPairs']
 
 
+def transform_ec2_key_pairs(
+        key_pairs: list[dict[str, Any]],
+        region: str,
+        current_aws_account_id: str,
+) -> list[dict[str, Any]]:
+    transformed_key_pairs = []
+    for key_pair in key_pairs:
+        key_name = key_pair["KeyName"]
+        transformed_key_pairs.append({
+            'KeyPairArn': f'arn:aws:ec2:{region}:{current_aws_account_id}:key-pair/{key_name}',
+            'KeyName': key_name,
+            'KeyFingerprint': key_pair.get("KeyFingerprint"),
```

Comment:
Are there every situations where we wouldn't see a fingerprint? My intuition is no

Reply:
And as such maybe we shouldn't use a `.get` so that we fail loudly?

Reply:
Sometimes yeah. AWS APIs can be inconsistent and sometimes return None on mandatory fields. Since fingerprint isn't used as a key anywhere, it's ok for it to be null. If it as for the keyname or arn then yeah we should fail loudly and not use .get().

describe-ec2-instances returns keypair information from a given ec2 instance and that [only returns an arn and keyname](https://github.com/cartography-cncf/cartography/blob/f08d85223b40b9f25a2a4ab6b26327e3d5b79a43/cartography/models/aws/ec2/keypair_instance.py#L17-L21), so having a fingerprint is an extra bit of information that only describe-key-pairs knows.

Reply:
There's also the analysis job to attach keypairs to other keypairs, and mark them as user-uploaded.
 * https://github.com/cartography-cncf/cartography/blob/aacaba9fa1b4f74db8eb6fcb8c3e6dd4b5588809/cartography/data/jobs/analysis/aws_ec2_keypair_analysis.json#L19-L28

Aside, We can probably build this functionality into NodeSchema.

Reply:
Another aside, I'm wondering if there is a good general purpose way to abstract it so that an analysis job writer doesn't have to write the cleanup step.